src-tauri/src/commands/mining.rs

```diff
@@ -108,89 +112,71 @@ fn get_proof_zero() -> Result<VDFProof, Error> {
-        return Ok(b)
-      }
-    match commit_proof::commit_proof_tx(&tx_params, b.clone(), false) {
-      Ok(tx_view) => match eval_tx_status(tx_view) {
-        Ok(_) => Ok(b),
-        Err(e) => {
-          let msg = format!(
-            "ERROR: Tower proof NOT committed to chain, message: \n{:?}",
-            e
-          );
-          println!("{}", &msg);
-          Err(CarpeError::tower(&msg))
-        }
-      },
-      Err(e) => {
-        let msg = format!("Tower transaction rejected, message: \n{:?}", e);
-        println!("{}", &msg);
-        Err(CarpeError::tower(&msg))
-      }
-    }
-  },
-    Err(e) => {
-      let msg = format!("Error mining tower proof, message: {:?}", e);
-      println!("{}", &msg);
-      Err(CarpeError::tower(&msg))
-    }
+#[tauri::command(async)]
+pub fn get_local_height() -> Result<u64, CarpeError> {
+  let cfg = get_cfg()?;
+  let block_dir = cfg.workspace.node_home.join(cfg.workspace.block_dir);
+  match parse_block_height(&block_dir).0 {
+      Some(h) => Ok(h),
+      None => Err(CarpeError::tower("could not get block height", TowerError::NoLocalBlocks.value())),
   }
 }
 
-// TODO: Resubmit backlog
-
-
-
-#[tauri::command]
+#[tauri::command(async)]
 pub fn get_local_proofs() -> Result<Vec<PathBuf>, CarpeError> {
-
   get_local_proofs_this_profile()
-  // TODO: Why is the CarpeError From anyhow not working?
-  .map_err(|e| { CarpeError::misc(&format!("could not get local files, message: {:?}", e.to_string()) ) })
+    // TODO: Why is the CarpeError From anyhow not working?
+    .map_err(|e| {
+      CarpeError::misc(&format!(
+        "could not get local files, message: {:?}",
+        e.to_string()
+      ))
+    })
+}
+#[derive(Debug, Clone, Serialize, Deserialize)]
+/// the parameter e.g. upper and lower thresholds
+pub struct EpochRules {
+  pub lower: u64,
+  pub upper: u64,
+  pub difficulty: u64,
+  pub security: u64,
 }
 
-#[tauri::command]
+#[tauri::command(async)]
+pub fn get_epoch_rules() -> Result<EpochRules, CarpeError> {
+  Ok(EpochRules {
+      lower: tower::EPOCH_MINING_THRES_LOWER,
+      upper: tower::EPOCH_MINING_THRES_UPPER,
+      difficulty: 0, // TODO: get from chain
+      security: 0, // TODO: get from chain
+  })
+}
+
+#[tauri::command(async)]
 pub fn set_env(env: String) -> Result<String, CarpeError> {
   match env.as_ref() {
     "test" => env::set_var("NODE_ENV", "test"),
     "prod" => env::set_var("NODE_ENV", "prod"),
-    _ => {},
+    _ => {}
   }
 
-  let v = env::var("NODE_ENV").map_err(|_| { CarpeError::misc("environment variable NODE_ENV is not set") })?;
+  let v = env::var("NODE_ENV")
+    .map_err(|_| CarpeError::misc("environment variable NODE_ENV is not set"))?;
   Ok(v)
 }
 
-#[tauri::command]
+#[tauri::command(async)]
 pub fn get_env() -> Result<String, CarpeError> {
-  let v = env::var("NODE_ENV").map_err(|_| { CarpeError::misc("environment variable NODE_ENV is not set") })?;
+  let v = env::var("NODE_ENV")
```

Comment:
I am prefer to return a prod as default value.  if so, we will reduce unnecessary error handling code.

Reply:
OK done.